css-conditional-5/Overview.bs

```diff
@@ -1067,6 +1087,159 @@ Style Container Features</h3>
 	are [=computed value|computed=] with respect to the [=query container=],
 	the same as other values.
 
+<h3 id="scroll-state-container">
+Scroll State Container Features</h3>
+
+	A <dfn export>container scroll-state query</dfn> allows to query a container for various
+	state that depends on scroll position. The ''scroll-state'' [=query container=] can either
+	be the scroller itself, or an element that is affected by the scroll position of an ancestor
+	[=scroll container's=] [=scrollport=].
+
+<h4 id="updating-scroll-state">
+Updating Scroll State</h4>
+	<div class=note>
+		Note: This section is subject to change as a result of resolving
+		<a href="https://github.com/w3c/csswg-drafts/issues/10796">issue #10796</a>
+	</div>
+
+	Scroll state may cause layout cycles since queried scroll state may cause style changes,
+	which may lead to scroll state changes as a result of layout. The same issue exists for
+	[=scroll progress timelines=], and scroll state is handled in a similar manner.
+
+	To avoid such layout cycles, ''scroll-state'' [=query containers=] update their
+	current state once as the last step of [=run the scroll steps=]. Then, after the
+	resizeObserver loop in the
+	<a href="https://html.spec.whatwg.org/multipage/webappapis.html#event-loop-processing-model">HTML event loop processing model</a>,
+	the scroll state of every ''scroll-state'' [=query container=] is updated.
+	If that state has changed since the scroll state update in [=run the scroll steps=],
+	re-run the style and layout update if necessary.
+
+<h4 id="stuck">
+Sticky positioning: the '@container/stuck' feature</h4>
+
+	<pre class="descdef mq">
+		Name: stuck
+		For: @container
+		Value: top | right | bottom | left | block-start | inline-start | block-end | inline-end
+		Type: discrete
+	</pre>
+
+	The '@container/stuck' [=container feature=] queries whether a
+	''position/sticky'' positioned container is visually shifted to stay inside
+	the [=sticky view rectangle=] for the given edge. The logical edges map to
+	physical based on the direction and writing-mode of the [=query container=].
+
+	In the boolean context, the query matches if visual shift is applied in any
```

Comment:
I assume that if the query container is not `position: sticky` then none of the values match?  It would be good to state that explicitly.  (It might make sense to have a bit less formal definition for each value in the `<dl>` since they have a lot in common.)

I also think that "visually shifted" should perhaps link to the definition in css-position-3.

It might be worth noting that a container can be stuck to two opposite sides at the same time.  (At least, after reading the text in css-position-3, I think that's the case.)

Reply:
> I assume that if the query container is not `position: sticky` then none of the values match? It would be good to state that explicitly. (It might make sense to have a bit less formal definition for each value in the `<dl>` since they have a lot in common.)

Done.

> I also think that "visually shifted" should perhaps link to the definition in css-position-3.

"visually shifted" does not have a dfn in that spec.

> It might be worth noting that a container can be stuck to two opposite sides at the same time. (At least, after reading the text in css-position-3, I think that's the case.)

Done.

Reply:
It's still not clear what happens when the sticky algorithm shifts both sides of the container in the same axis.  It's OK to address that as a followup, though.

Reply:
> It's still not clear what happens when the sticky algorithm shifts both sides of the container in the same axis. It's OK to address that as a followup, though.

How is that even possible? It can only be shifted in one direction per axis, right?

Reply:
Ah, never mind.  I guess I missed a sentence earlier in the [spec's text](https://drafts.csswg.org/css-position-3/#stickypos-insets):

> If this results in a [sticky view rectangle](https://drafts.csswg.org/css-position-3/#sticky-view-rectangle) size in any axis less than the size of the [border box](https://drafts.csswg.org/css-box-4/#border-box) of the sticky box in that axis, then the effective [end](https://drafts.csswg.org/css-writing-modes-4/#css-end)-edge inset in the affected axis is reduced (possibly becoming negative) to bring the sticky view rectangle’s size up to the size of the border box in that axis (where end is interpreted relative to the [writing mode](https://drafts.csswg.org/css-writing-modes-4/#writing-mode) of the [containing block](https://drafts.csswg.org/css-display-4/#containing-block)).

that removes the possibility that this can happen.  (But otherwise, reading just the algorithm in the following paragraph, it seems like it could... although I admit it's not clear what the priority would be!)